xlators/mgmt/glusterd/src/snapshot/glusterd-zfs-snapshot.c

```diff
@@ -0,0 +1,482 @@
+/*
+   Copyright (c) 2021 iXsystems, Inc <https://www.ixsystems.com>
+   This file is part of GlusterFS.
+
+   This file is licensed to you under your choice of the GNU Lesser
+   General Public License, version 3 or any later version (LGPLv3 or
+   later), or the GNU General Public License, version 2 (GPLv2), in all
+   cases as published by the Free Software Foundation.
+*/
+
+#include <inttypes.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "glusterd-messages.h"
+#include "glusterd-errno.h"
+
+#include "glusterd.h"
+#include "glusterd-utils.h"
+#include "glusterd-snapshot-utils.h"
+
+#include <glusterfs/dict.h>
+#include <glusterfs/run.h>
+
+#if defined(GF_LINUX_HOST_OS)
+#include <mntent.h>
+#else
+#include "mntent_compat.h"
+#endif
+
+#define ZFS_COMMAND "/usr/sbin/zfs"
+#define ZPOOL_COMMAND "/usr/sbin/zpool"
```

Comment:
Latest one uses only `#define ZFS_COMMAND "/sbin/zfs"` (https://github.com/gluster/glusterfs/pull/2855/files#diff-e56fa01b9d9da20ebddaf48db7ad09e7d89900cf72aafd840943e26f3cc66024R31)

> Is this OK to hardcode this path?

That needs some work. I used `glusterd_is_cmd_available` to check the zfs command is available. That is expecting absolute path. Same hardcode path is used for LVM as well. `/sbin/zfs` path works well with standard installation in Ubuntu/Debian/CentOS.

Can this be a separate PR or do I need to fix the issue with this PR?

> should we take this from configure?

Not possible since this is a external program.

Reply:
> Can this be a separate PR or do I need to fix the issue with this PR?

Sure. Please add a todo for now.